wallet/wallet-cli-lib/tests/basic.rs

```diff
@@ -0,0 +1,234 @@
+        .register(mempool.clone().into_rpc())
+        .register(p2p.clone().into_rpc())
+        .build()
+        .await
+        .unwrap();
+    let rpc_http_address = rpc.http_address().cloned().unwrap();
+    manager.add_subsystem("rpc", rpc);
+
+    (manager, rpc_http_address)
+}
+
+struct CliTestFramework {
+    rpc_address: SocketAddr,
+    shutdown_trigger: ShutdownTrigger,
+    manager_task: ManagerJoinHandle,
+    test_root: TestRoot,
+}
+
+impl CliTestFramework {
+    async fn setup() -> Self {
+        // logging::init_logging::<std::path::PathBuf>(None);
+
+        let test_root = test_utils::test_root!("wallet-cli-tests").unwrap();
+        // let test_dir = test_root.fresh_test_dir("basic_wallet_cli");
+
+        let (manager, rpc_address) = start_node().await;
+
+        let shutdown_trigger = manager.make_shutdown_trigger();
+        let manager_task = manager.main_in_task();
+
+        Self {
+            manager_task,
+            shutdown_trigger,
+            rpc_address,
+            test_root,
+        }
+    }
+
+    async fn run(&self, commands: &[&str]) -> Vec<String> {
+        let wallet_options = WalletCliArgs {
+            network: Network::Regtest,
+            wallet_file: None,
+            rpc_address: Some(self.rpc_address),
+            rpc_cookie_file: None,
+            rpc_username: Some(RPC_USERNAME.to_owned()),
+            rpc_password: Some(RPC_PASSWORD.to_owned()),
+            commands_file: None,
+            history_file: None,
+            exit_on_error: None,
+            vi_mode: false,
+        };
+
+        let console = MockConsole::new(commands);
+        wallet_cli_lib::run(console.clone(), wallet_options).await.unwrap();
+        let res = console.output.lock().unwrap().clone();
+        res
+    }
+
+    async fn shutdown(self) {
+        self.shutdown_trigger.initiate();
+        self.manager_task.join().await;
+        self.test_root.delete();
+    }
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn wallet_cli_basic() {
+    let test = CliTestFramework::setup().await;
+
+    let output = test.run(&["nodeversion"]).await;
+    assert_eq!(output, vec!["0.1.0"]);
+
+    let output = test.run(&["bestblockheight"]).await;
+    assert_eq!(output, vec!["0"]);
+
+    test.shutdown().await;
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn wallet_cli_file() {
+    let test = CliTestFramework::setup().await;
+
+    // Use dir name with spaces to make sure quoting works as expected
+    let file_name = test
+        .test_root
+        .fresh_test_dir("wallet dir")
+        .as_ref()
+        .join("wallet1")
+        .to_str()
+        .unwrap()
+        .to_owned();
+
+    let output = test.run(&[&format!("createwallet \"{file_name}\"")]).await;
+    assert_eq!(output.len(), 1);
+    assert!(output[0].starts_with("New wallet created successfully\n"));
+
+    let output = test.run(&[&format!("openwallet \"{file_name}\""), "closewallet"]).await;
```

Comment:
If I understand this correctly, please add a comment "Start the wallet, open it, then close it, then shutdown".